mlir/include/mlir/Dialect/MIOpen/Generator/Conv2dGenerator.h

```diff
@@ -127,6 +126,8 @@ class Conv2dGenerator {
     return permutation;
   }
   int getBwdDataKernelCount() const;
+  LogicalResult isValidDimension() const;
```

Comment:
My one complaint is that it feels like the name should be `hasValidDimension()` (and also `hasValidChip()` below) since those make the code more grammatical for me. That is, "if is applicable" makes sense (and its out of context "if generator is applicable", but not "if generator is valid dimension", which I'd prefer to be "if generator has valid dimension"